### ISSUE TITLE
URL generation error for Custom API Endpoint

### ISSUE THREAD

**Issue by hippothomas** [altcha-org/wordpress-plugin#13]
Hello,

I have an issue with the selfhosted API, my WordPress API is not on the same base url as the WordPress site url, so challenge url used by the plugin is not pointing to the right url.

My API url is `https://mydomain.com/wp-json/`
And the `get_site_url()` Wordpress function return `https://mydomain.com/wp/`
So the generated url in `AltchaPlugin->get_challengeurl()` is `https://mydomain.com/wp/wp-json/altcha/v1/challenge`

To prevent that, would it be possible to add a filter in the return of the `get_challengeurl` function ? That way we could easily change the default behaviour

Thanks!
ps: I can do a PR if you want

**Comment by ovx** [comment]
Hi, there's an options to change to the `REST API prefix` in the plugin's settings. If you change it to `wp/wp-json` it should work.

Late I'll also add the filter and improve the function to auto-detect this configuration.

**Comment by hippothomas** [comment]
The issue is that my api is on `/wp-json/` and not `/wp/wp-json/`, so the prefix option actually don't work in my case because the default prefix is actually the right one.
The issue comes from the url from `get_site_url()` which is pointing to my back-end who do not share the same URL with my API.
It's the default behavior for WordPress installs using [bedrock](https://roots.io/bedrock/).

**Comment by ovx** [comment]
Okay now I get it. It should use `get_rest_url` function instead and it will work automatically;

```php
public function get_challengeurl()
  {
    $api = $this->get_api();
    if ($api === "selfhosted") {
      return get_rest_url(null, "/altcha/v1/challenge");
    }
    $api_key = $this->get_api_key();
    return "https://$api.altcha.org/api/v1/challenge?apiKey=$api_key";
  }
```

I'll update the plugin soon.

**Comment by hippothomas** [comment]
This would be perfect :)
Thank you for your reactivity!

**Comment by ovx** [comment]
Fixed in version `1.5.0`.
